### PR TITLE
add --fill_empty flag to archive_database_update

### PR DESCRIPTION
**Pull request by BradleySappington**
Fill the gaps from the archive_database_update.py update

### CODE REVIEW

**Comment by BradleySappington** [comment]
@mfixstsci  AFTER MERGING TODO
On the production database:
1. Comment out the cronjob that runs this script: `41 * * * * /home/svc_jwqladm_ops/setup.sh /home/svc_jwqladm_ops/jwql/jwql/jwql/website/apps/jwql/archive_database_update.py >> /grp/jwst/ins/jwql/logs/test_archive_db_update_log.txt 2>&1`
2. run the script to update incomplete proposals using category as the empty field`$ python archive_database_update.py --fill_empty proposal category`
3. when complete run the script to update incomplete rootfileinfos using expstart as the empty field: `python archive_database_update.py --fill_empty rootfileinfo expstart`
4. when complete, re-enable cronjob on production.

**Comment by BradleySappington** [comment]
@bhilbert4 @melanieclarke @mfixstsci - Ready for review.  Currently not loaded on test or dev, so can still be tested on either database when one opens up.

File: jwql/website/apps/jwql/archive_database_update.py
Comment (melanieclarke) [inline jwql/website/apps/jwql/archive_database_update.py]: Trying this example exactly returns with an error: model_name incorrect.   The model should be 'rootfileinfo'

Comment (BradleySappington) [reply jwql/website/apps/jwql/archive_database_update.py]: fixed!

Comment (melanieclarke) [reply jwql/website/apps/jwql/archive_database_update.py]: Thanks!  I ran this example locally, and it found 9082 changes to make, but the database didn't seem to be changing.  It looks like there are large numbers of files that actually don't have filter values -- all the MIRI IFU files and at least some FGS files have no FILTER keyword in the FITS headers.

I just canceled that run and ran with expstart instead of filter.  That appropriately found 23 FGS files that needed updating, and fixed them.

So, maybe one more change to the example, to try expstart instead of filter?  But the code looks like it's working!

Comment (BradleySappington) [reply jwql/website/apps/jwql/archive_database_update.py]: Thats great feedback.  I remembered to put the specific fields to use in this PR, but I should also probably include that tidbit in the actual docstring.

Comment (BradleySappington) [reply jwql/website/apps/jwql/archive_database_update.py]: And done

Comment (melanieclarke) [reply jwql/website/apps/jwql/archive_database_update.py]: Thanks!

**Review by melanieclarke — APPROVED** [review]

**Comment by mfixstsci** [comment]
Thank you for testing @melanieclarke and thank you for the changes @BradleySappington I will go ahead and run the commands above.

**Comment by mfixstsci** [comment]
> Thank you for reviewing @melanieclarke and thank you for the changes @BradleySappington I will go ahead and run the commands above.

**Comment by mfixstsci** [comment]
@BradleySappington sorry about closing, that was an accident 🙄 waiting for CI to run and then I will merge it.

**Pull request merged**
